bobcat/bb-cumber/src/main/java/com/cognifide/qa/bb/cumber/Bobcumber.java

```diff
@@ -80,66 +80,49 @@
    */
   public Bobcumber(Class<?> clazz) throws InitializationError, IOException {
     super(clazz);
+    bindFields();
     storeFailedResults = clazz.isAnnotationPresent(StoreFailedResults.class);
     if (storeFailedResults) {
-      file = new File(clazz.getAnnotation(StoreFailedResults.class).value());
-      if (!file.exists()) {
-        PrintWriter writer = new PrintWriter(file, CharEncoding.UTF_8);
-        writer.close();
-      }
-      if (RERUN_FAILED_TESTS_CLASS_NAME.equals(clazz.getName()) && isFeatureFileEmpty()) {
-        noTestsToRerun = true;
-      }
+      featureFile = createFile(clazz.getAnnotation(StoreFailedResults.class).value());
+      statisticsFile = createFile(statisticsFilePath);
     }
+    isItFailedTestsRerun = clazz.isAnnotationPresent(FailedTestsRunner.class);
   }
 
   @Override
   public void run(RunNotifier notifier) {
-    if (noTestsToRerun) {
-      notifier.fireTestFinished(Description.EMPTY);
-      return;
+    if (isItFailedTestsRerun) {
+      try {
+        if (statisticsUtils.getNumberOfFailedTests(statisticsFile) == 0) {
+          notifier.fireTestFinished(Description.EMPTY);
+          return;
+        }
+        double percentageOfFailedTests = statisticsUtils.getPercentageOfFailedTests(statisticsFile);
+        if (percentageOfFailedTests > maxFailedTestPercentage) {
+          notifier.fireTestFailure(new Failure(
+              Description.createSuiteDescription(
+                  "Percentage of failed tests was bigger than " + maxFailedTestPercentage + "."),
+              new TooManyTestsException("Percentage of failed tests was bigger than " + maxFailedTestPercentage + ".")));
+          return;
+        }
+      } catch (IOException e) {
+        LOG.error("Statistics file not found.");
```

Comment:
log the exception and provide more generic info. This is not only reason that results in thorwing IOException